lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -403,52 +472,61 @@ sub get_test_results {
 
     eval { $translator->data } if $translator;    # Provoke lazy loading of translation data
 
-    my $test_info = $self->{db}->test_results( $params->{id} );
+    my $test_info;
     my @zm_results;
-    foreach my $test_res ( @{ $test_info->{results} } ) {
-        my $res;
-        if ( $test_res->{module} eq 'NAMESERVER' ) {
-            $res->{ns} = ( $test_res->{args}->{ns} ) ? ( $test_res->{args}->{ns} ) : ( 'All' );
-        }
-        elsif ($test_res->{module} eq 'SYSTEM'
-            && $test_res->{tag} eq 'POLICY_DISABLED'
-            && $test_res->{args}->{name} eq 'Example' )
-        {
-            next;
-        }
-
-        $res->{module} = $test_res->{module};
-        $res->{message} = $translator->translate_tag( $test_res, $params->{language} ) . "\n";
-        $res->{message} =~ s/,/, /isg;
-        $res->{message} =~ s/;/; /isg;
-        $res->{level} = $test_res->{level};
-
-        if ( $test_res->{module} eq 'SYSTEM' ) {
-            if ( $res->{message} =~ /policy\.json/ ) {
-                my ( $policy ) = ( $res->{message} =~ /\s(\/.*)$/ );
-                if ( $policy ) {
-                    my $policy_description = 'DEFAULT POLICY';
-                    $policy_description = 'SOME OTHER POLICY' if ( $policy =~ /some\/other\/policy\/path/ );
-                    $res->{message} =~ s/$policy/$policy_description/;
-                }
-                else {
-                    $res->{message} = 'UNKNOWN POLICY FORMAT';
-                }
+    eval{
+        $test_info = $self->{db}->test_results( $params->{id} );
+        foreach my $test_res ( @{ $test_info->{results} } ) {
+            my $res;
+            if ( $test_res->{module} eq 'NAMESERVER' ) {
+                $res->{ns} = ( $test_res->{args}->{ns} ) ? ( $test_res->{args}->{ns} ) : ( 'All' );
+            }
+            elsif ($test_res->{module} eq 'SYSTEM'
+                && $test_res->{tag} eq 'POLICY_DISABLED'
+                && $test_res->{args}->{name} eq 'Example' )
+            {
+                next;
             }
-            elsif ( $res->{message} =~ /config\.json/ ) {
-                my ( $config ) = ( $res->{message} =~ /\s(\/.*)$/ );
-                if ( $config ) {
-                    my $config_description = 'DEFAULT CONFIGURATION';
-                    $config_description = 'SOME OTHER CONFIGURATION' if ( $config =~ /some\/other\/configuration\/path/ );
-                    $res->{message} =~ s/$config/$config_description/;
+
+            $res->{module} = $test_res->{module};
+            $res->{message} = $translator->translate_tag( $test_res, $params->{language} ) . "\n";
+            $res->{message} =~ s/,/, /isg;
```

Comment:
Is this meant to expand "," to ", " in an argument? But that will add an extra " " if there is a "," in the msgid, won't it?

Reply:
Outside of the scope of this PR